### ISSUE TITLE
Delete multiple blobs

### ISSUE THREAD

**Issue by elibarzilay** [Azure/azure-cli#3757]
IIUC, `az storage blob delete` will delete just one blob.

Similar to #998, it would be nice to be able to delete a whole
"directory".  It looks like the only way to do that currently is to get
a list of all blobs using `--prefix` then iterate over each one of them,
which would overall be extremely slow compared to a single
`delete-batch`.

**Comment by elibarzilay** [comment]
@tjprescott Is there any chance that this will get implemented?  We currently have a script that garbage-collects old build artifacts and since we added documentation pages the process of a single version deletion takes about 30 minutes...

**Comment by tjprescott** [comment]
Sounds like a very reasonable request, although if the service doesn't support batch deletion then anything we do would basically be the same loop you are doing behind but behind the scenes. @troydai, @mayurid for technical feasibility and priority.

**Comment by elibarzilay** [comment]
Well, one thing that would be different is that if the cli is doing the deletions in one call, then the implementation could do the iteration over each blob and issue a delete, all on a single connection --- and that would be a huge saving over what can be done now since doing the loop outside the cli means about 2 seconds for each request which is probably mostly spent on establishing a connection.

In other words, it would make it as fast deleting a "directory" with the storage explorer, which is still way faster than spending two seconds over each deleted blob...

**Comment by troydai** [comment]
https://docs.microsoft.com/en-us/rest/api/storageservices/delete-blob
Currently, there isn't a REST API call for batch deletion. We can implement batch delete for your convenience, but we basically do the same thing as looping blob in a script.

I'm not sure what optimization the storage explorer has done. I'll look into that.

**Comment by troydai** [comment]
Another option is multi-threading. Then the main topic will be how to do throttling and prevent racing situation.

**Comment by elibarzilay** [comment]
@troydai, I don't know what it actually does, but I think that a lot of
speedup can happen if you keep the connection alive -- and it seems to
me that this could lead to the speedup that happens when deleting a
subtree in the storage explorer.  Assuming that most of the ~2 second
per deletion via the cli is spent on establishing a connection, this
would be the boost that will make it a practical operation.

**Comment by troydai** [comment]
@elibarzilay agrees. Thanks for the advice. I'll see what we can do to improve the performance.

**Comment by elibarzilay** [comment]
Thanks!

**Comment by phdowling** [comment]
Is this going to be implemented? I have a pyspark job generating tons of files in blob storage, but since `sc.saveAsTextFile` does not allow overwriting, I'm stuck having to delete the files (or leave them and generate new directories all the time, or deleting the entire container). This is pretty inconvenient.

I would actually love a feature where you don't need to list files at all, but rather specify a folder, prefix or maybe a regex pattern of files to be deleted.

**Comment by troydai** [comment]
@phdowling thank you for "up voting" this issue :). Looks like popular concern. I'll add this issue to the next sprint. We generally ship every two to three weeks.

**Comment by elibarzilay** [comment]
@williexu -- Thanks!!  What's the time frame for this feature to be part of the available version?

**Comment by williexu** [comment]
@elibarzilay  you're welcome, sorry it took so long.
Here's a link to our schedule: https://github.com/Azure/azure-cli/milestones
Our next version release will be on 11/15

**Comment by elibarzilay** [comment]
@williexu -- thanks, I'll follow it (it's very relevant for me, since I do have these long deletion sequences that are very slow).
